wpengine/faustjs
# Reduce App Router bundle size

## Tasks


## Description

This PR reduces the bundle size of the experimental app router package by using the proper `ApolloClient` and `InMemoryCache` vs the [guide's ](https://www.apollographql.com/blog/announcement/frontend/using-apollo-client-with-next-js-13-releasing-an-official-library-to-support-the-app-router/#server-components) `NextSSRApolloClient`/`NextSSRInMemoryCache` utils. Reference [this issue](https://github.com/apollographql/apollo-client-nextjs/issues/95) for more context.

This PR also removes support for the CJS build as it is no longer needed with the ESM support in Next.js. 



## Related Issue(s):



## Testing

From the monorepo root:

1. `npm install`
2. `npm run build` to build the packages
3. Build the example project `npm run build -w examples/next/app-router`
4. Notice the bundle stats are 150kb+
5. Check out the `app-router-fix-bundle` branch
6. `npm install`
7. `npm run build`
8. Build the example project again `npm run build -w examples/next/app-router`
9. Notice the bundle stats are much more reasonable (~80kb)



## Screenshots

Before:
<img width="994" alt="Screenshot 2023-09-18 at 8 26 31 AM" src="https://github.com/wpengine/faustjs/assets/5946219/b08127dc-8e7c-4f86-9a0a-4d5696205fca">

After:
<img width="1030" alt="Screenshot 2023-09-18 at 8 25 18 AM" src="https://github.com/wpengine/faustjs/assets/5946219/c14ddc40-5575-49b9-8316-a7747d000d13">




## Documentation Changes



## Dependant PRs
